src/gtest/test_merkletree.cpp

```diff
@@ -55,32 +56,46 @@ void expect_ser_test_vector(B& b, const C& c, const A& tree) {
 }
 
 template<typename Tree, typename Witness>
-void test_tree(Array root_tests, Array ser_tests, Array witness_ser_tests, Array path_tests) {
+void test_tree(
+    Array commitment_tests,
+    Array root_tests,
+    Array ser_tests,
+    Array witness_ser_tests,
+    Array path_tests
+)
+{
+    Array::iterator commitment_iterator = commitment_tests.begin();
     Array::iterator root_iterator = root_tests.begin();
     Array::iterator ser_iterator = ser_tests.begin();
     Array::iterator witness_ser_iterator = witness_ser_tests.begin();
     Array::iterator path_iterator = path_tests.begin();
 
-    uint256 test_commitment = uint256S("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855");
-
     Tree tree;
 
     // The root of the tree at this point is expected to be the root of the
     // empty tree.
     ASSERT_TRUE(tree.root() == Tree::empty_root());
 
+    // The tree doesn't have a 'last' element added since it's blank.
+    ASSERT_THROW(tree.last(), std::runtime_error);
+
     // We need to witness at every single point in the tree, so
     // that the consistency of the tree and the merkle paths can
     // be checked.
     vector<Witness> witnesses;
 
     for (size_t i = 0; i < 16; i++) {
```

Comment:
Since it's not involved in this PR I will leave it alone, but it's the number of elements that can fit in the testing tree (depth 4), so a constant already exists for it (`INCREMENTAL_MERKLE_TREE_DEPTH_TESTING`).